generated/nidaqmx/_grpc_interpreter.py

```diff
@@ -3362,14 +3394,15 @@ def get_error_string(self, error_code):
             return 'Failed to retrieve error description.'
 
 
-def _assign_numpy_array(numpy_array, grpc_array):
+def _assign_numpy_array(numpy_array, grpc_array, samples_read):
     """
     Assigns grpc array to numpy array maintaining the original shape.
 
     Checks for the instance of grpc_array with bytes, if validated to True,
     the numpy array is assigned to a 1D array of the grpc arrray.
     """
     grpc_array_size = len(grpc_array)
+    grpc_array[samples_read:] = numpy_array[samples_read:]
```

Comment:
It looks like you're trying to preserve the unread part of the numpy array by copying it to the grpc array and then copying it back to the numpy array. This approach might happen to fix the one failing test, but it's wrong.
- It modifies the wrong array. A function named `_assign_numpy_array` should assign to the numpy array. It should modify the source array.
- I doubt that it works with `bytes` or 2D arrays.

The non-destructive way to do this is: `grpc_array_size = min(len(grpc_array), samples_read)`

However, this is still wrong. The array contains `(num_channels * samps_per_chan)` elements, not `samps_per_chan` elements.

Can you get the info you need from the numpy array's shape? I don't know. Looking at https://github.com/ni/nidaqmx-python/blob/master/generated/nidaqmx/stream_readers.py#L72 there are 3 supported array shapes:

```
        if is_many_chan:
            if is_many_samp:
                array_shape = (number_of_channels,
                               number_of_samples_per_channel)
            else:
                array_shape = (number_of_channels,)
        else:
            if is_many_samp:
                array_shape = (number_of_samples_per_channel,)
```

Both `(number_of_channels,)` and `(number_of_samples_per_channel,)` are 1D arrays, so I don't think you can distinguish between them based on the shape of the array.

Reply:
I think all of the affected read functions have a `num_samps_per_chan` parameter. It should be possible to calculate `num_channels` based on the array size and `num_samps_per_chan`, but be careful about the corner cases (`num_samps_per_chan == 0`, `read_array.size % num_samps_per_chan != 0`). Also make sure to use integer division (`//`) not floating point division (`/`).

Reply:
The current version of this code still modifies the gRPC array